### DISCUSSION TITLE
IntersectionObserver is not defined

### DISCUSSION THREAD

**Discussion by Leila-ad** [taiga-family/taiga-ui#3944] [Q&A] [answered]
Hi I used the Taiga UI editor and I can't fix unit test error "IntersectionObserver is not defined".... I have read https://taiga-ui.dev/testing/jest this page and tried it and instead it gives me "jest worker encountered 4 child process exceptions, exceeding retry limit" error. :( how can I fix it 
Thank you in advance

**Comment by splincode** [comment]
Hello please provide your github repo for reproduction

**Comment by Leila-ad** [comment]
Hello,
thank you for your fast response, I have made this repository just to test.
https://github.com/LeSrbz/taiga-editor.git
CLI for testing: nx run texteditor:test
Thank you

**Comment by splincode** [comment]
In your file:

<img width="548" alt="image" src="https://user-images.githubusercontent.com/12021443/226872428-f9a2d78a-e4f5-4b20-9f1b-5e658ccf2323.png">


Don't forget 

```ts

import 'jest-preset-angular/setup-jest';
import '@ng-web-apis/universal/mocks';

window.IntersectionObserver = function () {
  this.observe = () => {};
  this.unobserve = () => {};
  this.disconnect = () => {};
};
```

In our docs

<img width="1030" alt="image" src="https://user-images.githubusercontent.com/12021443/226872793-d0dd3e40-a218-4c2a-b691-3d516bdff5cf.png">

**Reply by waterplea** [reply]
I believe `@ng-web-apis/universal/mocks` contains that mock for `IntersectionObserver`. If not, we should add it.

**Comment by Leila-ad** [comment]
I did that but then it gives another error. did you try it?

**Reply by waterplea** [reply]
Please use replies and not post new answers :)

**Reply by Leila-ad** [reply]
opps sorry was replying through email 😄

**Comment by waterplea** [comment, marked as answer]
This should not happen on the latest versions
